assets/src/components/ladder.tsx

```diff
@@ -0,0 +1,190 @@
+import React from "react"
+import { Timepoint, TimepointId, Vehicle } from "../skate"
+
+export interface Props {
+  timepoints: Timepoint[]
+  vehicles: Vehicle[]
+  ladderDirection: LadderDirection
+}
+
+// Timepoints come from the API in the ZeroToOne direction
+export enum LadderDirection {
+  ZeroToOne,
+  OneToZero,
+}
+
+export const flipLadderDirection = (
+  ladderDirection: LadderDirection
+): LadderDirection =>
+  ladderDirection === LadderDirection.ZeroToOne
+    ? LadderDirection.OneToZero
+    : LadderDirection.ZeroToOne
+
+const HEIGHT = 500
+const WIDTH = 180
+const CENTER_TO_LINE = 40 // x-distance between the center of the ladder and the center of the line
+const CENTER_TO_VEHICLE = 60 // x-distance between the center of the ladder and the center of the vehicles
+const MARGIN_TOP_BOTTOM = 40 // space between the top of the route and the top of the viewbox
+
+enum VehicleDirection {
+  Up,
+  Down,
+}
+
+const Timepoint = ({ timepoint, cy }: { timepoint: Timepoint; cy: number }) => (
+  <>
+    <circle
+      className="m-ladder__stop-circle"
+      cx={-CENTER_TO_LINE}
+      cy={cy}
+      r="3"
+    />
+    <circle
+      className="m-ladder__stop-circle"
+      cx={CENTER_TO_LINE}
+      cy={cy}
+      r="3"
+    />
+    <text
+      className="m-ladder__timepoint-name"
+      x="0"
+      y={cy}
+      textAnchor="middle"
+      dominantBaseline="middle"
+    >
+      {timepoint.id}
+    </text>
+  </>
+)
+
+interface VehicleProps {
+  vehicle: Vehicle
+  timepoints: Timepoint[]
+  timepointSpacingY: number
+  ladderDirection: LadderDirection
+}
+
+const Vehicle = ({
+  vehicle,
+  timepoints,
+  timepointSpacingY,
+  ladderDirection,
+}: VehicleProps) => {
+  const vehicleDirection: VehicleDirection =
+    (vehicle.direction_id === 1) ===
+    (ladderDirection === LadderDirection.ZeroToOne)
+      ? VehicleDirection.Down
+      : VehicleDirection.Up
+  const x =
+    vehicleDirection === VehicleDirection.Up
+      ? CENTER_TO_VEHICLE
+      : -CENTER_TO_VEHICLE
+  const y =
+    yForVehicle(vehicle, timepoints, timepointSpacingY, vehicleDirection) || -10
```

Comment:
What if `yForVehicle` return `-10` instead of `null` so all that logic was encapsulated there?

Reply:
This `|| -10` is just a temporary hack until we figure out the right thing to do with buses without a timepoint. The pending box? Don't draw them at all? Put them in the center-top instead of the same column as all the others? This default decision doesn't quite fit in with the "find the timepoint and adjust from there" logic.

Perhaps the `yForVehicle` name should be changed to reflect that it's more about finding the y of the vehicle's `timepoint_status` than the be-all-end-all of vehicle positioning?

Reply:
This sounds like open work that we haven't accounted for yet. Do you happen to have any examples of buses without timepoints that we could use as examples we could categorize and turn into tickets?

Reply:
[Followup Task](https://app.asana.com/0/1112935048846093/1122859757950933)